File: src/main/EscapeFromTheAliensInOuterSpace.java
//package main;
//
//import javafx.application.Application;
//import javafx.scene.Scene;
//import javafx.scene.control.TabPane;
//import javafx.scene.layout.HBox;
//import javafx.stage.Stage;
//import model.map.MapConfiguration;
//import model.map.MapNode;
//import model.CharacterType;
//import model.engine.GameClient;
//import model.Player;
//import model.player.PlayerCharacter;
//import view.GameView;
//import view.HostServerView;
//import view.JoinServerView;
//import view.SocialView;
//
//public class EscapeFromTheAliensInOuterSpace extends Application{
//	private static final int WIDTH = 1320;
//	private static final int HEIGHT = 1000;
//	private static final int INTRO_FRAME_HEIGHT = 300;
//	private static final int INTRO_FRAME_WIDTH = 400;
//	private static final String TITLE = "Escape from the Aliens in Outer Space";
//	private static final String VERSION_NUMBER = "v0.1";
//	private static Stage primaryStage;
//
//	private static GameClient client;
//
//
//	public static void main(String[] args){
//		launch(args);
//	}
//
//	@Override
//	public void start(Stage stage){
//		primaryStage = stage;
//		stage.setTitle(TITLE + " - " + VERSION_NUMBER);
//		stage.setResizable(false);
//		stage.setScene(renderConnectToServerWindow());
//		stage.setOnCloseRequest(e -> System.exit(0));
//		stage.show();
//	}
//
//
//	public static void createGameClient(String serverAddress, int portNumber, String username){
//		primaryStage.hide();
//		client = new GameClient(initializeLocalPlayer(username), serverAddress, portNumber);
//		new Thread(client).start();
//		primaryStage.setScene(renderGameWindow());
//		primaryStage.show();
//	}
//
//	public static Scene renderConnectToServerWindow(){
//		TabPane tabPane = new TabPane();
//		tabPane.getTabs().addAll(new JoinServerView(), new HostServerView());
//		Scene scene = new Scene(tabPane, INTRO_FRAME_WIDTH, INTRO_FRAME_HEIGHT);
//        return scene;
//	}
//
//	public static Scene renderGameWindow(){
//		HBox root = new HBox();
//		//MapConfiguration cfg = MapConfiguration.loadFromFile("src/main.resources/fermi.ser");
//		//MapConfiguration cfg = MapConfiguration.loadFromFile("src/main.resources/galilei.ser");
//		MapConfiguration cfg = MapConfiguration.readConfigurationFromFile(EscapeFromTheAliensInOuterSpace.class.getResourceAsStream("resources/galilei.ser"));
//		GameMap(cfg);
//		root.getChildren().add(new GameView(client));
//		root.getChildren().add(new SocialView(client));
//		Scene scene = new Scene(root, WIDTH, HEIGHT);
//		scene.getStylesheets().add(EscapeFromTheAliensInOuterSpace.class.getResource("resources/stylesheet.css").toExternalForm());
//		return scene;
//	}
//
//	private static Player initializeLocalPlayer(String username){
//		//TODO: Change the way that characters are created (probably more beneficial once the game lobby is established)
//		PlayerCharacter pc = new PlayerCharacter(CharacterType.ALIEN, MapNode.get("A03"));
//		return new Player(pc, username);
//	}
//}

File: src/main/EscapeFromTheAliensInOuterSpace.scala
package main

import javafx.application.Application
import javafx.scene.Scene
import javafx.scene.control.TabPane
import javafx.scene.layout.HBox
import javafx.stage.Stage

import model.map._
import model.player.PlayerCharacter
import net.{GameClient, GameServer}
import view.GameView
import view.HostServerView
import view.JoinServerView
import view.SocialView

object EscapeFromTheAliensInOuterSpace {
  val WIDTH: Int = 1320
  val HEIGHT: Int = 1000
  val INTRO_FRAME_HEIGHT: Int = 300
  val INTRO_FRAME_WIDTH: Int = 400
  val TITLE: String = "Escape from the Aliens in Outer Space"
  val VERSION_NUMBER: String = "v0.1"

  def main(args: Array[String]) {
    Application.launch(classOf[EscapeFromTheAliensInOuterSpace], args: _*)
  }

  def createGameClient(
    serverAddress: String,
    portNumber: Int,
    primaryStage: Stage
  ): Unit = {
      primaryStage.hide()
      primaryStage.setScene(GameClient.createGameClient(serverAddress, portNumber))
      primaryStage.show()
  }

  def createGameServer(
    serverAddress: String,
    portNumber: Int,
    primaryStage: Stage
  ): Unit = {
    // TODO: Actually create the server!
    GameServer.createServer(portNumber)

    createGameClient(serverAddress, portNumber, primaryStage)
  }

  def renderConnectToServerWindow(stage: Stage): Scene = {
    val tabPane: TabPane = new TabPane
    tabPane.getTabs.addAll(new JoinServerView(stage), new HostServerView(stage))
    val scene: Scene = new Scene(tabPane, INTRO_FRAME_WIDTH, INTRO_FRAME_HEIGHT)
    scene
  }

//  private def initializeLocalPlayer(username: String): Player = {
//    //TODO: Change the way that characters are created (probably more beneficial once the game lobby is established)
//    val pc: PlayerCharacter = new PlayerCharacter(CharacterType.ALIEN, MapNode.get("A03"))
//    new Player(pc, username)
//  }
}

class EscapeFromTheAliensInOuterSpace extends Application {
  def start(stage: Stage) {
    stage.setTitle(EscapeFromTheAliensInOuterSpace.TITLE + " - " + EscapeFromTheAliensInOuterSpace.VERSION_NUMBER)
    stage.setResizable(false)
    stage.setScene(EscapeFromTheAliensInOuterSpace.renderConnectToServerWindow(stage))
    stage.setOnCloseRequest(e => sys.exit(0))
    stage.show()
  }
}